src/pages/index.js

```diff
@@ -40,19 +42,51 @@ const previewImageCloseButton = document.querySelector(
 const profileEditPopup = new PopupWithForm(
   "#profile-edit-modal",
   (formData) => {
-    userInfo.setUserInfo({
-      name: formData["name"],
-      title: formData["title"],
-    });
-    profileEditPopup.close();
+    api
+      .updateUserProfile({
+        name: formData.name,
+        about: formData.about,
+      })
+      .then(() => {
+        userInfo.setUserInfo({
+          name: formData.name,
+          about: formData.about,
+        });
+        profileEditPopup.close();
+      })
+      .catch((err) => {
+        console.error(`ERROR UPDATING USER PROFILE ${err}`);
+      });
+  }
+);
+
+const editAvatarPopup = new PopupWithForm(
+  "#update-avatar-modal",
+  (formData) => {
+    api
+      .updateAvatar({ avatar: formData.avatar })
+      .then((userData) => {
+        userInfo.setUserAvatar({ avatar: userData.avatar });
+        editAvatarPopup.close();
+      })
+      .catch((err) => {
+        console.error(`ERROR UPDATING AVATAR ${err}`);
+      });
   }
 );
 
+// editAvatarPopup.setEventListeners();
+avatarEditButton.addEventListener("click", () => {
+  editAvatarPopup.open();
+});
+
+editAvatarPopup.setEventListeners();
+
 addNewCardButton.addEventListener("click", () => newCardPopup.open());
 document.querySelector("#profile-edit-button").addEventListener("click", () => {
-  const { name, title } = userInfo.getUserInfo();
+  const { name, about } = userInfo.getUserInfo();
   profileNameInput.value = name;
-  profileTitleInput.value = title;
+  profileTitleInput.value = about;
   profileEditPopup.open();
 });
 
```

Comment:
Please, delete ` this.disableButton();` from line 81 in `FormValidator.js`

You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.